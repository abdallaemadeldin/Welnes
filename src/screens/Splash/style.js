import { StyleSheet } from 'react-native';

export const style = () => {
    return StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: '#ccc',
            justifyContent: 'center',
            alignItems: 'center'
        },
        logo: {
            width: '80%',
            resizeMode: 'center'
        }
    });
}