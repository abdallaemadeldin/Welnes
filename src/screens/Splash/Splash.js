import React, { memo } from 'react';
import { View, Image } from 'react-native';
import { useSplash } from 'src/hooks';
import { style } from './style';

const Splash = () => {
    const { container, logo } = style();
    useSplash();

    return (
        <View style={container}>
            <Image source={require('src/assets/logo.png')} style={logo} />
        </View>
    );
}

export default memo(Splash);