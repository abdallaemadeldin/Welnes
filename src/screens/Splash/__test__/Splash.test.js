import React from 'react';
import renderer from 'react-test-renderer';
import Splash from '../Splash';

jest.mock("react-native-fs", () => "RNFS");
jest.mock("react-native-image-crop-picker", () => "RNImagePicker");
jest.mock("react-native-ffmpeg", () => "FFMPEG");

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            replace: jest.fn(),
        }),
    };
});

it('renders correctly', () => {
    const tree = renderer.create(<Splash />).toJSON();
    expect(tree).toMatchSnapshot();
});