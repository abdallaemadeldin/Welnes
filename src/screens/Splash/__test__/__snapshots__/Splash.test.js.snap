// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ccc",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logo.png",
      }
    }
    style={
      Object {
        "resizeMode": "center",
        "width": "80%",
      }
    }
  />
</View>
`;
